catalogs/catalog-fileset/src/main/java/org/apache/gravitino/catalog/fileset/FilesetCatalogOperations.java

```diff
@@ -137,6 +140,23 @@ public class FilesetCatalogOperations extends ManagedSchemaOperations
   @VisibleForTesting ScheduledThreadPoolExecutor scheduler;
   @VisibleForTesting Cache<FileSystemCacheKey, FileSystem> fileSystemCache;
 
+  private final ThreadPoolExecutor fileSystemExecutor =
+      new ThreadPoolExecutor(
+          Math.max(2, Runtime.getRuntime().availableProcessors() * 2),
```

Comment:
This will create too many threads for server if they're more than 32 cores.

Reply:
I limit the max thread number to 24